source/common/upstream/subset_lb.cc

```diff
@@ -110,6 +112,62 @@ void SubsetLoadBalancer::refreshSubsets(uint32_t priority) {
   update(priority, host_sets[priority]->hosts(), {});
 }
 
+void SubsetLoadBalancer::initSubsetSelectorMap() {
+  SubsetSelectorMapPtr selectors;
+  for (const auto& subset_selector : subset_selectors_) {
+    const auto& selector_keys = subset_selector.first;
+    unsigned long pos = 0;
+    selectors = selectors_;
+    for (const auto& key : selector_keys) {
+      const auto& selector_it = selectors->subset_keys.find(key);
```

Comment:
This must be crashing some unit tests as selectors_ is default initialized only.

Reply:
thanks, fixed, previous version of code looked like this:
<code>
struct SubsetSelectorMap {
    std::unordered_map<std::string, SubsetSelectorMap> subset_keys;
    ...
  }
;</code>
and clang compiler on my machine did not complain about `incomplete type` but build failed on CI. Switched to smart pointers in `SubsetSelectorMap` struct instead.

Reply:
Sorry again, I'm unsure how I have missed selectors_ initialization on the first line of this function. Please disregard this comment.

Reply:
@yanavlasov selectors_ initialization was not there indeed, apologies i had to do force push to correct DCO.